zypp/zyppng/base/private/threaddata_p.h

```diff
@@ -5,38 +5,40 @@
 #include <thread>
 #include <string>
 
-namespace zyppng {
-
+namespace zyppng
+{
   class EventDispatcher;
 
-  struct ThreadData {
-    static ThreadData &current ();
+  struct ThreadData
+  {
+    static ThreadData &current();
 
     template<typename T>
-    void setName ( T &&name ) {
-      threadName = std::forward<T>( name );
+    void setName( T &&name ) {
+      _threadName = std::forward<T>( name );
       syncNativeName();
     }
 
-    std::string name () const;
-
-    std::shared_ptr<EventDispatcher> ensureDispatcher ();
-    void setDispatcher (const std::shared_ptr<EventDispatcher> &disp );
+    const std::string &name() const;
 
+    std::shared_ptr<EventDispatcher> dispatcher()
+    { return _dispatcher.lock(); }
+    std::shared_ptr<EventDispatcher> ensureDispatcher();
+    void setDispatcher( const std::shared_ptr<EventDispatcher> &disp );
 
-    std::weak_ptr<EventDispatcher> dispatcher;
-    std::thread::id threadId;
-    std::string threadName;
 
   private:
-    void syncNativeName ();
+    void syncNativeName();
     ThreadData();
 
   private:
+    std::thread::id _threadId;
```

Comment:
But you IMO should. It's a valuable information for everybody new to the class and also once you refactor it. You simply know that you don't have to worry about the item, because it does not escape the class.